### PR TITLE
Use Service Monitor

### PR DESCRIPTION
**Pull request by QuentinBisson**
This Configures this app to use a servicemonitor

### CODE REVIEW

**Comment by QuentinBisson** [comment]
image

File: helm/cluster-api-provider-azure/templates/rbac.authorization.k8s.io_v1_rolebinding_capz-leader-election-rolebinding.yaml [hunk 1]
```diff
@@ -9,6 +9,8 @@ metadata:
     app.kubernetes.io/managed-by: '{{ .Release.Service }}'
     app.kubernetes.io/name: cluster-api-provider-azure
     app.kubernetes.io/version: '{{ .Chart.AppVersion }}'
+    application.giantswarm.io/team: '{{ index .Chart.Annotations "application.giantswarm.io/team"
+      }}'
```
Comment (whites11) [inline helm/cluster-api-provider-azure/templates/rbac.authorization.k8s.io_v1_rolebinding_capz-leader-election-rolebinding.yaml:13]: is the new line intended?

Comment (QuentinBisson) [reply helm/cluster-api-provider-azure/templates/rbac.authorization.k8s.io_v1_rolebinding_capz-leader-election-rolebinding.yaml:13]: I think it's a side effect of make

**Comment by QuentinBisson** [comment]
@whites11 Can I go ahead with this one ?

**Comment by QuentinBisson** [comment]
Could someone from @giantswarm/team-phoenix take a look please?

**Comment by weseven** [comment]
@QuentinBisson can we make this optional through a helm value like in https://github.com/giantswarm/cluster-api-provider-aws-app/pull/245?
In the bootstrap cluster used by mc-bootstrap there are of course no prometheus CRDs, and currently the install of the latest version of the provider apps including the serviceMonitor fails. 

The value `serviceMonitor.enabled=false` seems already configured in the script doing the helm install in mc-bootstrap, it's just missing the implementation on the provider charts: https://github.com/giantswarm/mc-bootstrap/blob/d3cda5e26dc83f33aeb8d905885e1e5506632019/scripts/setup-capi.sh#L47-L48

**Comment by QuentinBisson** [comment]
Yes we can

**Comment by weseven** [comment]
Done :)

**Pull request merged**